drivers/display/display_st7528i.c

```diff
@@ -0,0 +1,505 @@
+
+	uint16_t height;
+	uint16_t width;
+
+	uint8_t framerate;
+	uint8_t com_partial_display;
+	uint8_t com_offset;
+	bool invert_com;
+	bool invert_segments;
+
+	uint8_t lcd_bias;
+	uint8_t regulator_resistor;
+	uint8_t electronic_volume;
+	uint8_t boost;
+};
+
+struct st7528i_data {
+	uint8_t framebuffer[ST7528_NUM_PAGES * ST7528_BPP * ST7528_WIDTH];
+};
+
+static int st7528i_write_cmds(const struct device *dev, const uint8_t *data, uint32_t len)
+{
+	const struct st7528i_config *config = dev->config;
+
+	return i2c_burst_write_dt(&config->bus, ST7528I_CONTROL_ALL_BYTES_CMD, data, len);
+}
+
+static int st7528i_write_data(const struct device *dev, const uint8_t *data, uint32_t len)
+{
+	const struct st7528i_config *config = dev->config;
+
+	return i2c_burst_write_dt(&config->bus, ST7528I_CONTROL_ALL_BYTES_DATA, data, len);
+}
+
+static int st7528i_set_mode(const struct device *dev, uint8_t mode)
+{
+	const struct st7528i_config *config = dev->config;
+	uint8_t cmd_buf[] = {
+		ST7528_CMD_SET_MODE,
+		ST7528_MODE_BOOSTER_EFFICIENCY_2 | (config->framerate << 4) | mode,
+	};
+
+	return st7528i_write_cmds(dev, cmd_buf, ARRAY_SIZE(cmd_buf));
+}
+
+static int st7528i_write_palette(const struct device *dev)
+{
+	int ret;
+
+	ret = st7528i_set_mode(dev, ST7528_MODE_PALETTE);
+	if (ret) {
+		return ret;
+	}
+
+	for (int index = 0; index < ST7528_NUM_GRAY_LEVELS; index++) {
+		uint8_t value = ST7528_PWM_LEVELS - (index * ST7528_PWM_LEVELS / 15);
+		uint8_t cmd = ST7528_CMD_SET_PALETTE_ENTRY | (index << 2);
+
+		uint8_t cmd_buf[] = {
+			/*
+			 * Same value for all four "frames" - not entirely clear from the datasheet
+			 * what that even means
+			 */
+			/* clang-format off */
+			cmd | 0x00, value,
+			cmd | 0x01, value,
+			cmd | 0x02, value,
+			cmd | 0x03, value,
+			/* clang-format on */
+		};
+
+		ret = st7528i_write_cmds(dev, cmd_buf, ARRAY_SIZE(cmd_buf));
+		if (ret) {
+			return ret;
+		}
+	}
+
+	return st7528i_set_mode(dev, ST7528_MODE_NORMAL);
+}
+
+static int st7528i_reset(const struct device *dev)
+{
+	const struct st7528i_config *config = dev->config;
+	int ret;
+
+	if (config->reset.port) {
+		gpio_pin_set_dt(&config->reset, true);
+		k_sleep(ST7528_RESET_DELAY);
+		gpio_pin_set_dt(&config->reset, false);
+		k_sleep(ST7528_RESET_DELAY);
+	}
+
+	{
+		uint8_t cmd_buf[] = {
+			/* clang-format off */
+			ST7528_CMD_SET_PARTIAL_DISPLAY_LINES, config->com_partial_display,
+			ST7528_CMD_ADC_SELECT | config->invert_segments,
+			ST7528_CMD_SHL_SELECT | (config->invert_com << 3),
+			ST7528_CMD_SET_INITIAL_COM0, config->com_offset,
+
+			ST7528_CMD_START_OSCILLATOR,
+			ST7528_CMD_SET_REGULATOR_RESISTOR | config->regulator_resistor,
+			ST7528_CMD_SET_ELECTRONIC_VOLUME, config->electronic_volume,
+			ST7528_CMD_SET_LCD_BIAS | (config->lcd_bias - 5),
+
+			ST7528_CMD_SET_BOOST | 0x00,
+			/* clang-format on */
+		};
+
+		ret = st7528i_write_cmds(dev, cmd_buf, ARRAY_SIZE(cmd_buf));
+		if (ret) {
+			return ret;
+		}
+	}
```

Comment:
still NACK to it, as in previous attempt.

Reply:
@jfischer-no couuld you please elaborate? I couldn't find any reference to an issue in a "previous attempt", assuming you're referring to PR #53596? 
If the issue is with `ARRAY_SIZE()`, it doesn't seem that uncommon and is used in a similar fashion in other instances in the codebase. As this seems to be the only outstanding issue with this PR, where a lot of work has obviously gone into, I hope this can get resolved :)
Thanks!